truss-build/src/truss/build/image.py

```diff
@@ -0,0 +1,165 @@
+from .commands import *
+from typing import Dict, List, Optional, Union
+from pathlib import Path
+from .utils import flatten_str_args, make_pip_install_args
+import shlex
+
+# TODO(bola): support secrets
+
+
+class Image:
+    def __init__(self) -> None:
+        self._base_image = ""
+        self._commands: list[Command] = []
+        self._base_image: Optional[str] = None
```

Comment:
Nit: To be very precise, class attributes should be type annotated before `__init__`.

```
class Image:
    _base_image: Optional[str]
    _commands: list[str]

    def __init__(self) -> None:
        ...
```

Reply:
when you say class attributes, this meand instance variables? They belong to `self` not `cls`, right?